images/controller/internal/controllers/rvr_diskful_count/reconciler_test.go

```diff
@@ -0,0 +1,613 @@
+package rvrdiskfulcount_test
+
+import (
+	"context"
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	v1alpha1 "github.com/deckhouse/sds-replicated-volume/api/v1alpha1"
+	v1alpha3 "github.com/deckhouse/sds-replicated-volume/api/v1alpha3"
+	rvrdiskfulcount "github.com/deckhouse/sds-replicated-volume/images/controller/internal/controllers/rvr_diskful_count"
+)
+
+func TestReconciler(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Reconciler Suite")
+}
+
+func newFakeClient() client.Client {
+	scheme := runtime.NewScheme()
+	_ = v1alpha1.AddToScheme(scheme)
+	_ = v1alpha3.AddToScheme(scheme)
+
+	return fake.NewClientBuilder().
+		WithScheme(scheme).
+		WithStatusSubresource(&v1alpha3.ReplicatedVolumeReplica{}, &v1alpha3.ReplicatedVolume{}).
+		Build()
+}
+
+func createReplicatedStorageClass(name, replication string) *v1alpha1.ReplicatedStorageClass {
+	return &v1alpha1.ReplicatedStorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: v1alpha1.ReplicatedStorageClassSpec{
+			Replication: replication,
+		},
+	}
+}
+
+func createReplicatedVolume(name, rscName string) *v1alpha3.ReplicatedVolume {
+	return &v1alpha3.ReplicatedVolume{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: v1alpha3.ReplicatedVolumeSpec{
+			Size:                       resource.MustParse("1Gi"),
+			ReplicatedStorageClassName: rscName,
+		},
+		Status: &v1alpha3.ReplicatedVolumeStatus{
+			Conditions: []metav1.Condition{},
+		},
+	}
+}
+
+func createReplicatedVolumeReplica(name, rvName string, ready bool, deletionTimestamp *metav1.Time) *v1alpha3.ReplicatedVolumeReplica {
+	rvr := &v1alpha3.ReplicatedVolumeReplica{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: "storage.deckhouse.io/v1alpha3",
+					Kind:       "ReplicatedVolume",
+					Name:       rvName,
+					Controller: func() *bool { b := true; return &b }(),
+				},
+			},
+		},
+		Spec: v1alpha3.ReplicatedVolumeReplicaSpec{
+			ReplicatedVolumeName: rvName,
+			Type:                 "Diskful",
+		},
+	}
+
+	if deletionTimestamp != nil {
+		rvr.DeletionTimestamp = deletionTimestamp
+	}
+
+	if ready {
+		rvr.Status = &v1alpha3.ReplicatedVolumeReplicaStatus{
+			Conditions: []metav1.Condition{
+				{
+					Type:   v1alpha3.ConditionTypeReady,
+					Status: metav1.ConditionTrue,
+				},
+			},
+		}
+	}
+
+	return rvr
+}
+
+var _ = Describe("Reconciler", func() {
+	var cl client.Client
+	var rec *rvrdiskfulcount.Reconciler
+	var ctx context.Context
+
+	BeforeEach(func() {
+		ctx = context.Background()
```

Comment:
```suggestion
	BeforeEach(func(ctx SpecContext) {
```
 Проверь чтоб везде так было

Reply:
готово!